server/src/alda/lisp/attributes.clj

```diff
@@ -69,6 +82,22 @@
   :initial-val 0.5
   :transform percentage)
 
+(defn- validate-str-key-sig
+  "Validates the current key-sig. Checks for:
+
+  1. No duplicate letters, ie: a- a+
+  2. No letters out of range a-g
+
+  If all tests pass, return true"
+  [key-sig]
+  ;; Get a version of key-sig with only characters
+  (let [clean-str (apply str (filter #(Character/isLetter %) key-sig))]
+    (and
+     ;; Check to see if any chars are not a-g
+     (empty? (re-find #"[^a-gA-G]" clean-str))
```

Comment:
1. We should actually only allow `a-g`. Something like `"F+ C+ G+"` should throw an error.

2. `re-find` will return `nil` if it doesn't find any instances of the regex in the string. So, it would be more idiomatic to use `not` here, instead of `empty?` which has slightly different semantics.

3. This is also a matter of personal preference, but I would include the first condition on the same line of the `and`, like this:

    ```clojure
    (and (not (re-find #"[^a-g]" clean-str))
         (= (count (distinct clean-str)) (count clean-str)))
    ```